api/gossiper.cc

```diff
@@ -57,12 +57,13 @@ void set_gossiper(http_context& ctx, routes& r) {
     });
 
     httpd::gossiper_json::assassinate_endpoint.set(r, [](std::unique_ptr<request> req) {
+        abort_source as;
```

Comment:
This variable is on the stack, below you pass it by reference to asynchronous functions which keep it after this function returns a future. I think you need to use some existing abort source, or do_with()?